code/game/objects/items/toys.dm

```diff
@@ -500,23 +487,72 @@
 	resistance_flags = FLAMMABLE
+	var/list/poof_sound = list('sound/weapons/thudswoosh.ogg' = 1)
+	var/stuffed = TRUE //If the plushie has stuffing in it
+	var/obj/item/grenade/grenade //You can remove the stuffing from a plushie and add a grenade to it for *nefarious uses*
 
 /obj/item/toy/plushie/attack(mob/M as mob, mob/user as mob)
-	playsound(loc, poof_sound, 20, 1)	// Play the whoosh sound in local area
+	playsound(loc, pickweight(poof_sound), 20, 1)	// Play the whoosh sound in local area
 	if(iscarbon(M))
 		if(prob(10))
 			M.reagents.add_reagent("hugs", 10)
 	return ..()
 
 /obj/item/toy/plushie/attack_self(mob/user as mob)
-	var/cuddle_verb = pick("hugs","cuddles","snugs")
-	user.visible_message("<span class='notice'>[user] [cuddle_verb] [src].</span>")
-	playsound(get_turf(src), poof_sound, 50, 1, -1)
+	if(stuffed || grenade)
+		var/cuddle_verb = pick("hugs","cuddles","snugs")
+		user.visible_message("<span class='notice'>[user] [cuddle_verb] [src].</span>")
+		playsound(get_turf(src), poof_sound, 50, 1, -1)
+		if(grenade && !grenade.active)
+			add_attack_logs(user, user, "activated a hidden grenade in [src].", ATKLOG_MOST)
+			playsound(loc, 'sound/weapons/armbomb.ogg', 10, 1, -3)
+			addtimer(CALLBACK(src, PROC_REF(explosive_betrayal), grenade), rand(1, 3) SECONDS) //We call with grenade as arugment, so cutting the grenade out doesn't magically defuse it
+	else
+		to_chat(user, "<span class='notice'>You try to pet [src], but it has no stuffing. Aww...</span>")
 	return ..()
 
+/obj/item/toy/plushie/proc/explosive_betrayal(/obj/item/grenade/grenade)
+	var/obj/item/grenade/GG = grenade
+	GG.prime()
+
+/obj/item/toy/plushie/Destroy()
+	QDEL_NULL(grenade)
+	..()
+
+/obj/item/toy/plushie/attackby(obj/item/I, mob/living/user, params)
+	if(I.sharp)
+		if(!grenade)
+			if(!stuffed)
+				to_chat(user, "<span class='warning'>You already murdered it!</span>")
+				return
+			user.visible_message("[user] tears out the stuffing from [src]!", "<span class='notice'>You rip a bunch of the stuffing from [src]. Murderer.</span>")
+			I.play_tool_sound(src)
+			stuffed = FALSE
+		else
+			to_chat(user, "<span class='notice'>You remove the grenade from [src].</span>")
+			grenade.forceMove(get_turf(src))
+			user.put_in_hands(grenade)
+			grenade = null
+		return
+	if(istype(I, /obj/item/grenade))
+		if(stuffed)
+			to_chat(user, "<span class='warning'>You need to remove some stuffing first!</span>")
+			return
+		if(grenade)
+			to_chat(user, "<span class='warning'>[src] already has a grenade!</span>")
+			return
+		if(!user.drop_item())
+			to_chat(user, "<span class='warning'>[I] is stuck to you and cannot be placed into [src].</span>")
+			return
+		user.visible_message("<span class='warning'>[user] slides [grenade] into [src].</span>", \
+		"<span class='warning'>You slide [I] into [src].</span>")
+		I.forceMove(src)
+		grenade = I
+		add_attack_logs(user, user, "placed a hidden grenade in [src].", ATKLOG_ALMOSTALL)
+		return
+
```

Comment:
You might want to add a ..() somewhere to this attack_by proc

Reply:
Done